napari/_qt/qt_main_window.py

```diff
@@ -474,23 +475,20 @@ def _help_changed(self, event):
         """
         self._help.setText(event.text)
 
-    def _screenshot_dialog(self):
+    def _screenshot_dialog(self, *, default_name=""):
         """Save screenshot of current display with viewer, default .png"""
-        filename, _ = QFileDialog.getSaveFileName(
-            parent=self.qt_viewer,
-            caption='Save screenshot with viewer',
-            directory=self.qt_viewer._last_visited_dir,  # home dir by default
-            filter="Image files (*.png *.bmp *.gif *.tif *.tiff)",  # first one used by default
-            # jpg and jpeg not included as they don't support an alpha channel
+        dial = ScreenshotDialog(
+            self.screenshot, self, self.qt_viewer._last_visited_dir
         )
-        if (filename != '') and (filename is not None):
-            # double check that an appropriate extension has been added as the
-            # filter option does not always add an extension on linux and windows
-            # see https://bugreports.qt.io/browse/QTBUG-27186
-            image_extensions = ('.bmp', '.gif', '.png', '.tif', '.tiff')
-            if not filename.endswith(image_extensions):
-                filename = filename + '.png'
-            self.screenshot(path=filename)
+        while True:
+            res = dial.exec_()
```

Comment:
This may be personal preference, but do we have to use the `while True` and `res = dial.exec_()` syntax? Can we make the dialog modal, (blocking) and then show it, and catch the return value only when it is closed?

Reply:
recursion done